test/integration/work/manifestworkreplicaset_test.go

```diff
@@ -193,6 +200,85 @@ var _ = ginkgo.Describe("ManifestWorkReplicaSet", func() {
+
+		works, err = hubWorkClient.WorkV1().ManifestWorks(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("work.open-cluster-management.io/manifestworkreplicaset=%s", key),
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		for _, work := range works.Items {
+			workCopy := work.DeepCopy()
+			meta.SetStatusCondition(&workCopy.Status.Conditions, metav1.Condition{Type: workapiv1.WorkApplied, Status: metav1.ConditionTrue, Reason: "ApplyTest"})
+			meta.SetStatusCondition(&workCopy.Status.Conditions, metav1.Condition{Type: workapiv1.WorkAvailable, Status: metav1.ConditionTrue, Reason: "ApplyTest"})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			_, err := hubWorkClient.WorkV1().ManifestWorks(workCopy.Namespace).UpdateStatus(context.TODO(), workCopy, metav1.UpdateOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		}
+		gomega.Eventually(assertWorksByReplicaSet(clusterNames, manifestWorkReplicaSet, 6), eventuallyTimeout, eventuallyInterval).Should(gomega.Succeed())
+	})
+
+	ginkgo.It("rolling deadline meets and tolerate max failure", func() {
+		manifestWorkReplicaSet, clusterNames, err := generateTestFixture(2, func(rs *workapiv1alpha1.ManifestWorkReplicaSet) *workapiv1alpha1.ManifestWorkReplicaSet {
+			rs.Spec.PlacementRefs[0].RolloutStrategy = clusterv1alpha1.RolloutStrategy{
+				Type: clusterv1alpha1.Progressive,
+				Progressive: &clusterv1alpha1.RolloutProgressive{
+					MaxConcurrency: intstr.FromInt32(1),
+					RolloutConfig: clusterv1alpha1.RolloutConfig{
+						ProgressDeadline: "5s",
+						MaxFailures:      intstr.FromInt32(1),
+					},
+				},
+			}
+			return rs
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		gomega.Eventually(assertWorksByReplicaSet(clusterNames, manifestWorkReplicaSet, 1), eventuallyTimeout, eventuallyInterval).Should(gomega.Succeed())
+
+		ginkgo.By("set work status to fail")
+		key := fmt.Sprintf("%s.%s", manifestWorkReplicaSet.Namespace, manifestWorkReplicaSet.Name)
+		works, err := hubWorkClient.WorkV1().ManifestWorks(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("work.open-cluster-management.io/manifestworkreplicaset=%s", key),
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		for _, work := range works.Items {
+			workCopy := work.DeepCopy()
+			meta.SetStatusCondition(&workCopy.Status.Conditions, metav1.Condition{Type: workapiv1.WorkApplied, Status: metav1.ConditionFalse, Reason: "ApplyTest"})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			_, err := hubWorkClient.WorkV1().ManifestWorks(workCopy.Namespace).UpdateStatus(context.TODO(), workCopy, metav1.UpdateOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("proceed to the next cluster after timeout")
+		gomega.Eventually(assertWorksByReplicaSet(clusterNames, manifestWorkReplicaSet, 2), eventuallyTimeout, eventuallyInterval).Should(gomega.Succeed())
+	})
```

Comment:
Could here add 2 more cases? One is "failure exceeds MaxFailures" and another is for `MinSuccessTime`.

Reply:
added